frontend/src/views/reading_view.js

```diff
@@ -78,23 +93,24 @@ class ReadingView extends React.Component {
             // Otherwise, move on to the rereading layout
             this.setState({rereading: true});
         }
-        window.scrollTo(0,0);
+        segment_ref.current.scrollTo(0,0);
     }
 
+    recordScroll = () => {
```

Comment:
Using class properties + arrow functions in this way to bind `this` is _super_ tempting because it makes the callback cleaner elsewhere, but alas it is not a very good idea for a bunch of subtle and annoying reasons pretty well documented [here](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) and [here](https://zendev.com/2018/10/01/javascript-arrow-functions-how-why-when.html) -- both articles worth reading now that you're into the guts of how this stuff works.

Alas, the better idea for now is to write this as a regular class method and then either explicitly do `this.recordScroll.bind(this)` in the constructor, or do `() => this.recordScroll()` whenever you need to pass it into a context where you need the `this` to point to the class.